libs/sr/formula-ui/src/char/StatBoostBonusAbility.tsx

```diff
@@ -0,0 +1,84 @@
+import type { AscensionKey } from '@genshin-optimizer/sr/consts'
+import type { Calculator } from '@genshin-optimizer/sr/formula'
+import { own } from '@genshin-optimizer/sr/formula'
+import { Translate } from '@genshin-optimizer/sr/i18n'
+import type {
+  TalentSheetElementBonusAbilityKey,
+  TalentSheetElementStatBoostKey,
+} from './consts'
+
+export type ReqMapEntry = {
+  subtitle: React.ReactNode
+  onClickable: (calc: Calculator) => boolean
+  disabled: (calc: Calculator) => boolean
+}
+function reqMapEntryAsc(
+  talentKey: TalentSheetElementStatBoostKey | TalentSheetElementBonusAbilityKey,
+  ascension: AscensionKey
+): ReqMapEntry {
+  return {
+    subtitle: (
+      <Translate ns="charSheet_gen" key18="req_asc" values={{ 1: ascension }} />
+    ),
+    onClickable: (calc: Calculator) =>
+      calc.compute(own.char.ascension).val >= ascension,
+    disabled: (calc: Calculator) =>
+      calc.compute(own.char.ascension).val < ascension ||
+      !calc.compute(own.char[talentKey]).val,
+  }
+}
+function reqMapEntryLvl(
+  talentKey: TalentSheetElementStatBoostKey | TalentSheetElementBonusAbilityKey,
+  level: number
+): ReqMapEntry {
+  return {
+    subtitle: (
+      <Translate ns="charSheet_gen" key18="req_lvl" values={{ 1: level }} />
+    ),
+    onClickable: (calc: Calculator) => calc.compute(own.char.lvl).val >= level,
+    disabled: (calc: Calculator) =>
+      calc.compute(own.char.lvl).val < level ||
+      !calc.compute(own.char[talentKey]).val,
+  }
+}
+
+export const bonusStatsReqMap: Record<
```

Comment:
Don't know if you wanna add a TODO for other dependencies (e.g. bonus stat X might rely on bonus stat X-3 being unlocked, and it changes per-path)

Reply:
It is datamineable (#2568)